logic/peer.js

```diff
@@ -103,7 +103,7 @@ Peer.prototype.parseInt = function (integer, fallback) {
 Peer.prototype.applyHeaders = function (headers) {
```

Comment:
my proposal is to avoid using `self` in future and to replace it with just this in every occurence. Bindings are used here anyway